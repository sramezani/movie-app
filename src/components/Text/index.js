import React from 'react';
import PropTypes from 'prop-types';
import {Text, StyleSheet, Platform} from 'react-native';

// Consts and Libs
import {Colors, AppFonts} from '../../theme';

/* Component =================================== */
function CustomText(props) {
    const _style = () => {
        let style = {
            ...AppFonts.base,
            color: Colors[props.color],
            fontSize: AppFonts[props.size].fontSize,
            fontWeight: AppFonts[props.weight],
        };

        // ---------------------- Decoration
        if (props.onPress) {
            style.textDecorationLine = 'underline';
        }

        // ---------------------- Paragraph
        if (props.p) {
            style.lineHeight = AppFonts[props.size].lineHeight * 1.35;
        }

        style = [style];

        if (props.style) style.push(props.style);

        return style;
    };

    return (
        <Text {...props} style={_style()}>
            {props.children}
        </Text>
    );
}

/* Styles ====================================== */
const styles = StyleSheet.create({});

/* Props ======================================= */
CustomText.propTypes = {
    size: PropTypes.oneOf([
        'xxxs',
        'xxs',
        'xs',
        'sm',
        'md',
        'lg',
        'xlg',
        'xxlg',
    ]),
    color: PropTypes.string,
    weight: PropTypes.oneOf(['ultraBold', 'bold', 'semiBold', 'normal']),

    p: PropTypes.bool, // paragraph

    onPress: PropTypes.func,
    style: PropTypes.oneOfType([
        PropTypes.array,
        PropTypes.number,
        PropTypes.object,
    ]),
    children: PropTypes.node,
};

CustomText.defaultProps = {
    size: 'sm',
    color: 'textColor',
    weight: 'normal',

    p: false,

    onPress: null,
    style: null,
    children: null,
};

/* Export Component ============================ */
export default CustomText;
