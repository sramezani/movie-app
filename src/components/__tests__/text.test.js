'use strict';

import 'react-native';
import React from 'react';
import Text from '../Text';
  
import renderer from 'react-test-renderer';

jest.setTimeout(15000);

test('renders correctly', () => {
  const tree = renderer.create(<Text size="sm" weight="bold" color="primary">hello</Text>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('check size', () => {
  const tree = renderer.create(<Text size="xxlg">hello</Text>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('check weight', () => {
  const tree = renderer.create(<Text weight="bold">hello</Text>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('check color', () => {
  const tree = renderer.create(<Text color="primary">hello</Text>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('check style', () => {
  const tree = renderer.create(<Text style={{ textAlign: 'center' }}>hello</Text>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('check onpress', () => {
  const tree = renderer.create(<Text onPress={() => console.log(123)}>hello</Text>).toJSON();
  expect(tree).toMatchSnapshot();
});