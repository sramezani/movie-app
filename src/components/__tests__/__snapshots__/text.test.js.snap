// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check color 1`] = `
<Text
  color="primary"
  onPress={null}
  p={false}
  size="sm"
  style={
    Array [
      Object {
        "color": "#039BE5",
        "fontSize": 14,
        "fontWeight": "400",
      },
    ]
  }
  weight="normal"
>
  hello
</Text>
`;

exports[`check onpress 1`] = `
<Text
  color="textColor"
  onPress={[Function]}
  p={false}
  size="sm"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 14,
        "fontWeight": "400",
        "textDecorationLine": "underline",
      },
    ]
  }
  weight="normal"
>
  hello
</Text>
`;

exports[`check size 1`] = `
<Text
  color="textColor"
  onPress={null}
  p={false}
  size="xxlg"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 28,
        "fontWeight": "400",
      },
    ]
  }
  weight="normal"
>
  hello
</Text>
`;

exports[`check style 1`] = `
<Text
  color="textColor"
  onPress={null}
  p={false}
  size="sm"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 14,
        "fontWeight": "400",
      },
      Object {
        "textAlign": "center",
      },
    ]
  }
  weight="normal"
>
  hello
</Text>
`;

exports[`check weight 1`] = `
<Text
  color="textColor"
  onPress={null}
  p={false}
  size="sm"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 14,
        "fontWeight": "700",
      },
    ]
  }
  weight="bold"
>
  hello
</Text>
`;

exports[`renders correctly 1`] = `
<Text
  color="primary"
  onPress={null}
  p={false}
  size="sm"
  style={
    Array [
      Object {
        "color": "#039BE5",
        "fontSize": 14,
        "fontWeight": "700",
      },
    ]
  }
  weight="bold"
>
  hello
</Text>
`;
